### PR TITLE
Audio PR - Augmentation support [ Resample, Operator overloading and Uniform + Normal Distribution nodes ]

### PR DESCRIPTION
**Pull request by SundarRajan98**
- Added node and python support for Resample, TensorAddTensor, TensorMulScalar, UniformDistribution and NormalDistribution augmentations
MIVisionX PR Link : PR https://github.com/ROCm/MIVisionX/pull/1323
Note: https://github.com/ROCm/rocAL/pull/125 should be merged before this PR - file changes will reduce to 23

### CODE REVIEW

**Comment by LakshmiKumar23** [comment]
@SundarRajan28 please resolve merge conflicts

File: rocAL/include/pipeline/batch_rng.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#pragma once
+#include <random>
+
+template <typename RNG = std::mt19937>
+class BatchRNG {
```
Comment (rrawther) [inline rocAL/include/pipeline/batch_rng.h:27]: where is this class used? why is this moved to a new file.

**Review by rrawther — COMMENTED** [review]
Please address review comments

File: rocAL/include/pipeline/batch_rng.h (hunk 1 above)
Comment (SundarRajan28) [reply rocAL/include/pipeline/batch_rng.h:27]: This class is used for uniform and normal distribution nodes. It is added to pipeline folder since this is required in both distribution nodes

Comment (rrawther) [reply rocAL/include/pipeline/batch_rng.h:27]: I suggest adding it to commons.h instead of creating a new file

**Review by rrawther — COMMENTED** [review]
Added a comment

**Comment by LakshmiKumar23** [comment]
@SundarRajan28 please resolve merge conflicts

File: rocAL/include/pipeline/batch_rng.h (hunk 1 above)
Comment (SundarRajan28) [reply rocAL/include/pipeline/batch_rng.h]: Moved class definition to commons.h

**Review by rrawther — APPROVED** [review]
@Tcherniaev, Pavel to unit test
